src/consensus/aft/async_execution.h

```diff
@@ -64,112 +79,137 @@ namespace aft
   {
   public:
     AppendEntryResponseCallback(
-      AbstractConsensusCallback& store_, AppendEntriesResponse&& hdr_) :
+      AbstractConsensusCallback& store_,
+      NodeId from_,
+      AppendEntriesResponse&& hdr_) :
       store(store_),
+      from(from_),
       hdr(std::move(hdr_))
     {}
 
     void execute() override
     {
-      store.recv_append_entries_response(hdr);
+      store.recv_append_entries_response(from, hdr);
     }
 
   private:
     AbstractConsensusCallback& store;
+    NodeId from;
     AppendEntriesResponse hdr;
   };
 
   class SignedAppendEntryResponseCallback : public AbstractMsgCallback
   {
   public:
     SignedAppendEntryResponseCallback(
-      AbstractConsensusCallback& store_, SignedAppendEntriesResponse&& hdr_) :
+      AbstractConsensusCallback& store_,
+      NodeId from_,
```

Comment:
Several `NodeId from_`s that could be `const NodeId& from_` in this file.

Reply:
Thanks! I thought I'd caught them all but clearly didn't. This should be fixed now.